collection_test.go

```diff
@@ -217,3 +217,24 @@ func TestPush(t *testing.T) {
 		t.Error("The keys were messed up :/")
 	}
 }
+
+func TestMerge(t *testing.T) {
+	collection := Collect[any](1, 2, "foo")
+
+	collectionToMerge := Collect[any](3, 4, "bar")
+
+	collection, err := collection.Merge(collectionToMerge)
+
+	if err != nil {
+		t.Error("The collection should not have returned an error")
+	}
+
+	for _, v := range []any{1, 2, "foo", 3, 4, "bar"} {
+
+		_, err := collection.Search(v)
```

Comment:
To ensure the keys weren't messed up, you could use the key returned by `Search` and pass to `Get`, comparing the resulting value. WDYT?